### PR TITLE
Smooth scrolling

### LINKED ISSUE #2713: Add smooth scrolling to CScrollRegion
Modern browsers like Chromium and Firefox scroll content smoothly.
We could use the same effect to make the scrolling less jarring.

### PR DESCRIPTION
**Pull request by TsFreddie**
closes #2713 

I hope this make scrolling feels better.

### CODE REVIEW

**Comment by oy** [comment]
The highlighted entry in the server browser flickers while scrolling.
Besides that it feels smooth.

**Comment by TsFreddie** [comment]
Yeah, it can't be avoided. its either the selection or the whole list. unless we want to animate selection change as well.

**Comment by TsFreddie** [comment]
We can disable hover highlight while the scroll region is animating. dont if thats confusing for user or not.

**Comment by oy** [comment]
Hm, that flickering wasn't there before. You really notice it, even when you scroll just 1 page. Something should be done about it.

**Comment by TsFreddie** [comment]
Try this.

Also made the scroll animation a little bit longer to make sure the highlight doesn't pop back a frame or two when you scroll the wheel in multiple bursts.

**Comment by Learath2** [comment]
Instead of hardcoding the curve right there you could take the cubic bezier we have in ddnet. Would be a nice reusable piece

**Comment by TsFreddie** [comment]
Doesn't seems worth it to add a class for this. Maybe a c function in `math.h`?

**Comment by Learath2** [comment]
Not like classes cost more, especially with modern compilers and no virtual functions

File: src/game/client/components/menus_scrollregion.cpp [hunk 1]
```diff
@@ -191,4 +226,9 @@ bool CMenus::CScrollRegion::IsRectClipped(const CUIRect& Rect) const
 bool CMenus::CScrollRegion::IsScrollbarShown() const
 {
 	return m_ContentH > m_ClipRect.h;
+}
+
+bool CMenus::CScrollRegion::IsAnimating() const
+{
+	return m_AnimTime > 0;
 }
```
Comment (oy) [inline src/game/client/components/menus_scrollregion.cpp:234]: Newline

File: src/game/client/components/menus_scrollregion.cpp [hunk 2]
```diff
@@ -88,11 +101,27 @@ void CMenus::CScrollRegion::End()
...
 		m_RequestScrollY = -1;
 	}
 
-	m_ScrollY = clamp(m_ScrollY, 0.0f, MaxScroll);
+	m_AnimTargetScrollY = clamp(m_AnimTargetScrollY, 0.0f, MaxScroll);
+
+	if (abs(m_AnimInitScrollY - m_AnimTargetScrollY) < 0.5f)
+		m_AnimTime = 0;
+
+	if (m_AnimTime > 0)
```
Comment (oy) [inline src/game/client/components/menus_scrollregion.cpp:114]: if(

File: src/game/client/components/menus_scrollregion.cpp [hunk 3]
```diff
@@ -88,11 +101,27 @@ void CMenus::CScrollRegion::End()
...
-		m_ScrollY = m_RequestScrollY;
+		m_AnimTargetScrollY = m_RequestScrollY;
+		m_AnimTime = 0;
 		m_RequestScrollY = -1;
 	}
 
-	m_ScrollY = clamp(m_ScrollY, 0.0f, MaxScroll);
+	m_AnimTargetScrollY = clamp(m_AnimTargetScrollY, 0.0f, MaxScroll);
+
+	if (abs(m_AnimInitScrollY - m_AnimTargetScrollY) < 0.5f)
```
Comment (oy) [inline src/game/client/components/menus_scrollregion.cpp:111]: if(
and `absolute`instead of `abs`

**Pull request merged**